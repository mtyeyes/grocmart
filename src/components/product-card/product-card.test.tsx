import * as React from 'react';
import { shallow } from 'enzyme';
import ProductCard, { Props } from './product-card';

const setUp = (props: Props) => shallow(<ProductCard {...props} />);

describe('ProductCard component', () => {
  const actionMock: any = () => {};
  const chilliPropsMock = {
    productId: 'chilli',
    productName: 'chilli peppers',
    priceBeforeDiscounts: '15',
    priceAfterDiscounts: '15',
    productRating: 3,
    addToCart: actionMock,
  };

  it('should render product card', () => {
    const component = setUp(chilliPropsMock);

    expect(component).toMatchSnapshot();
  });

  it('should render product card with "popular" flag', () => {
    const popularChilliProps = {
      ...chilliPropsMock,
      productRating: 5,
    };
    const component = setUp(popularChilliProps);

    const flag = component.find('.product-card__flag');
    const flagText = flag.text();

    expect(flagText).toMatch('popular');
  });

  it('should render product card with "on sale" flag', () => {
    const onSaleChillyProps = {
      ...chilliPropsMock,
      priceBeforeDiscounts: '15',
      priceAfterDiscounts: '12',
    };
    const component = setUp(onSaleChillyProps);

    const flag = component.find('.product-card__flag');
    const flagText = flag.text();

    expect(flagText).toMatch('sale');
  });
});
