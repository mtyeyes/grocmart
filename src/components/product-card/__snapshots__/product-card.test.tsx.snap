// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component should render product card 1`] = `
<li
  className="product-card__container"
  key="chilli"
>
  <div
    className="product-card"
  >
    <div
      className="product-card__flags-container"
    />
    <img
      alt="chilli peppers"
      className="product-card__thumbnail"
      loading="lazy"
      src="/grocmart/images/chilli.png"
    />
    <h4
      className="product-card__title"
    >
      chilli peppers
    </h4>
    <div
      className="product-card__price-container"
    >
      <p
        className="product-card__price product-card__price--total"
      >
        <HiddenText>
          Total price
        </HiddenText>
        15
      </p>
    </div>
    <div
      className="product-card__controls"
    >
      <LinkAsButton
        className="product-card__btn product-card__btn--product-info"
        subtype="round-grey"
        to="/shop/chilli"
      >
        <Icon
          iconId="search"
        />
        <HiddenText>
          Product info
        </HiddenText>
      </LinkAsButton>
      <Button
        className="product-card__btn product-card__btn--add-to-cart"
        onClick={[Function]}
      >
        <Icon
          iconId="basket"
        />
        <HiddenText>
          Add to cart
        </HiddenText>
      </Button>
    </div>
  </div>
</li>
`;
